lib/file/mkdir_windows.go

```diff
@@ -0,0 +1,78 @@
+//go:build windows
+// +build windows
+
+package file
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+// MkdirAll creates a directory named path, along with any necessary parents.
+//
+// Improves os.MkdirAll by avoiding trying to create a folder \\? when the
+// volume of a given extended length path does not exist.
+//
+// Based on source code from golang's os.MkdirAll, with improvements from
+// the source code of Syncthing, and also some additional improvements,
+// all changes marked by comments.
+// For comparsion, see:
+// https://github.com/syncthing/syncthing/blob/main/lib/fs/basicfs_windows.go
+// https://github.com/golang/go/blob/master/src/os/path.go
+func MkdirAll(path string, perm os.FileMode) error {
+	// Fast path: if we can tell whether path is a directory or file, stop with success or error.
+	dir, err := os.Stat(path)
+	if err == nil {
+		if dir.IsDir() {
+			return nil
+		}
+		return &os.PathError{
+			Op:   "mkdir",
+			Path: path,
+			Err:  syscall.ENOTDIR,
+		}
+	}
+
+	// Slow path: make sure parent exists and then call Mkdir for path.
+	i := len(path)
+	for i > 0 && os.IsPathSeparator(path[i-1]) { // Skip trailing path separator.
+		i--
+	}
+
+	// Change 1: The following two lines are additions compared to, not only os.MkdirAll, but also Syncthing's variant.
```

Comment:
Can you remove these `Change...` comments? and describe what the code does without reference to some other code which isn't here!

Reply:
Sure. Thought you would want to know exactly what/why was different from original, but if thats not important then definitely will make it cleaner.